web/service/src/main/java/org/mobilitydata/gtfsvalidator/web/service/Main.java

```diff
@@ -0,0 +1,233 @@
+      var objectMapper = new ObjectMapper();
+      var jobNode = objectMapper.readTree(fileBytes);
+      var countryCode = jobNode.get(countryCodeKey).textValue();
+      return countryCode;
+    } catch (Exception exc) {
+      System.out.println(
+          "Error could not load remote file, using default country code:" + exc.toString());
+      // Use default
+      return defaultCountryCode;
+    }
+  }
+
+  @CrossOrigin(origins = "*")
+  @PostMapping(value = "/create-job")
+  public String createJob(@RequestBody CreateJobBody body) {
+    try {
+      final var jobId = UUID.randomUUID().toString();
+      var countryCode = defaultCountryCode;
+
+      if (body != null && !body.getCountryCode().isEmpty()) {
+        countryCode = body.getCountryCode();
+      }
+
+      if (body != null && body.getUrl() != null && !body.getUrl().isEmpty()) {
+        // Read file into memory
+        var url = body.getUrl();
+        var urlInputStream = new BufferedInputStream(new URL(url).openStream());
+
+        // Upload to GCS
+        var blobId = BlobId.of(userUploadBucketName, jobId + "/" + jobId + ".zip");
+        var mimeType = "application/zip";
+        var blobInfo = BlobInfo.newBuilder(blobId).setContentType(mimeType).build();
+        var fileBytes = urlInputStream.readAllBytes();
+        storage.create(blobInfo, fileBytes);
+        setJobCountryCode(jobId, countryCode);
+        return "{\"jobId\": \"" + jobId + "\"}";
+      } else {
+        var blobInfo =
+            BlobInfo.newBuilder(BlobId.of(userUploadBucketName, jobId + "/" + fileName)).build();
+
+        // Generate Signed URL
+        Map<String, String> extensionHeaders = new HashMap<>();
+        extensionHeaders.put("Content-Type", "application/octet-stream");
+
+        URL url =
+            getStorage()
+                .signUrl(
+                    blobInfo,
+                    15,
+                    TimeUnit.MINUTES,
+                    Storage.SignUrlOption.httpMethod(HttpMethod.PUT),
+                    Storage.SignUrlOption.withExtHeaders(extensionHeaders),
+                    Storage.SignUrlOption.withV4Signature());
+
+        setJobCountryCode(jobId, countryCode);
+        return "{\"jobId\": \"" + jobId + "\", \"url\": \"" + url.toString() + "\"}";
+      }
+    } catch (Exception exc) {
+      System.out.println("Error:" + exc.toString());
+      throw new ResponseStatusException(HttpStatus.INTERNAL_SERVER_ERROR, "Error", exc);
+    }
+  }
+
+  @PostMapping("/run-validator")
+  public ResponseEntity runValidator(@RequestBody Body body) {
+    try {
+      var message = body.getMessage();
+      if (message == null) {
+        var msg = "Bad Request: invalid Pub/Sub message format";
+        System.out.println(msg);
+        return new ResponseEntity(msg, HttpStatus.BAD_REQUEST);
+      }
+
+      var data = new String(Base64.getDecoder().decode(message.getData()));
+
+      var map = new ObjectMapper();
+      var node = map.readTree(data);
```

Comment:
Ditto comment on introduce an explicit type for deserializing whatever this data looks like.